client/src/main/java/org/apache/rocketmq/client/impl/MQClientAPIImpl.java

```diff
@@ -577,12 +577,12 @@ public void operationComplete(ResponseFuture responseFuture) {
                     }
                 } else {
                     if (!responseFuture.isSendRequestOK()) {
-                        pullCallback.onException(new MQClientException("send request failed", responseFuture.getCause()));
+                        pullCallback.onException(new MQClientException("send request failed to " + addr + ".request: " + request, responseFuture.getCause()));
```

Comment:
How about writing it this way?
`new MQClientException("send request [" + request+ "] failed to " + addr, responseFuture.getCause())`